File: src/ROCrate/ISAProfile/LabProtocol.fs
namespace ARCtrl.ROCrate

open DynamicObj
open Fable.Core

///
[<AttachMembers>]
type LabProtocol(
    id,
    ?additionalType,
    ?name,
    ?intendedUse,
    ?description,
    ?url,
    ?comment,
    ?version,
    ?labEquipment,
    ?reagent,
    ?computationalTool
) as this =
    inherit LDObject(
        id = id,
        schemaType = ResizeArray[|"bioschemas.org/LabProtocol"|],
        additionalType = defaultArg additionalType (ResizeArray[||])
    )
    do
        DynObj.setOptionalProperty (nameof name) name                            this
        DynObj.setOptionalProperty (nameof intendedUse) intendedUse              this
        DynObj.setOptionalProperty (nameof description) description              this
        DynObj.setOptionalProperty (nameof url) url                              this
        DynObj.setOptionalProperty (nameof comment) comment                      this
        DynObj.setOptionalProperty (nameof version) version                      this
        DynObj.setOptionalProperty (nameof labEquipment) labEquipment            this
        DynObj.setOptionalProperty (nameof reagent) reagent                      this
        DynObj.setOptionalProperty (nameof computationalTool) computationalTool  this

File: js/ROCrate/ISAProfile/LabProtocol.js
import { defaultOf } from "../../fable_modules/fable-library-js.4.22.0/Util.js";
import { FSharpRef } from "../../fable_modules/fable-library-js.4.22.0/Types.js";
import { setOptionalProperty } from "../../fable_modules/DynamicObj.4.0.3/DynObj.fs.js";
import { LDObject_$reflection, LDObject } from "../LDObject.js";
import { unwrap } from "../../fable_modules/fable-library-js.4.22.0/Option.js";
import { class_type } from "../../fable_modules/fable-library-js.4.22.0/Reflection.js";

export class LabProtocol extends LDObject {
    constructor(id, additionalType, name, intendedUse, description, url, comment, version, labEquipment, reagent, computationalTool) {
        super(id, "bioschemas.org/LabProtocol", unwrap(additionalType));
        const this$ = new FSharpRef(defaultOf());
        this$.contents = this;
        this["init@8"] = 1;
        setOptionalProperty("name", name, this$.contents);
        setOptionalProperty("intendedUse", intendedUse, this$.contents);
        setOptionalProperty("description", description, this$.contents);
        setOptionalProperty("url", url, this$.contents);
        setOptionalProperty("comment", comment, this$.contents);
        setOptionalProperty("version", version, this$.contents);
        setOptionalProperty("labEquipment", labEquipment, this$.contents);
        setOptionalProperty("reagent", reagent, this$.contents);
        setOptionalProperty("computationalTool", computationalTool, this$.contents);
    }
}

export function LabProtocol_$reflection() {
    return class_type("ARCtrl.ROCrate.LabProtocol", undefined, LabProtocol, LDObject_$reflection());
}

export function LabProtocol_$ctor_3514295B(id, additionalType, name, intendedUse, description, url, comment, version, labEquipment, reagent, computationalTool) {
    return new LabProtocol(id, additionalType, name, intendedUse, description, url, comment, version, labEquipment, reagent, computationalTool);
}

